fastapi/applications.py

```diff
@@ -125,6 +131,16 @@ def __init__(
         self.openapi_schema: Optional[Dict[str, Any]] = None
         self.setup()
 
+    @property
+    def lifetime_dependencies(self) -> Dict[Tuple[Callable[..., Any], Tuple[str]], Any]:
+        if not hasattr(self, "_lifetime_dependencies"):
+            self._lifetime_dependencies = {}
+        return self._lifetime_dependencies
+
+    @lifetime_dependencies.setter
+    def lifetime_dependencies(self, value: Dict[Tuple[Callable[..., Any], Tuple[str]], Any]) -> None:
```

Comment:
What's the reason behind this getter and setter? Why not just assign the variable in the `__init__` method?

Reply:
Probably works as well, what I have here is over complicated and doesn't fully work.

Really I'm just struggling to come up with a good way to deal with binding the lifetime dependency stack `__aenter__`/`__aexit__` to the app lifetime. Do you have any suggestions in that regard?

Reply:
I completely reworked this part. I got rid of the setters/getters, and it is now handling lifespans as expected. However, I'm not convinced that creating 3 async closures in `__init__` and binding them to `on_start`/`on_shutdown` is the cleanest way to do this.